os_migrate/plugins/module_utils/network.py

```diff
@@ -136,6 +136,27 @@ def network_refs_from_sdk(conn, sdk_net):
     return refs
 
 
+def security_group_rule_refs_from_sdk(conn, sdk_rule):
+    """Create a dict of name/id mappings for resources referenced from
+    OpenStack SDK Security Group Rule `sdk_rule`. Fetch any necessary information
+    from OpenStack SDK connection `conn`.
+
+    Returns: dict with names and IDs of resources referenced from
+    `sdk_rule` (only those important for OS-Migrate)
+    """
+    expected_type = openstack.network.v2.security_group_rule.SecurityGroupRule
+    if type(sdk_rule) != expected_type:
+        raise exc.UnexpectedResourceType(expected_type, type(sdk_rule))
+    refs = {}
+
+    # when creating refs from SDK Security Group Rule object, we copy IDs and
+    # query the cloud for names
+    refs['security_group_name'] = reference.security_group_name(
+        conn, sdk_rule['security_group_id'])
```

Comment:
We should add remote_group_name from remote_group_id, using the same security_group_name lookup function.

Reply:
done